src/main/scala/uk/gov/nationalarchives/PuidRepository.scala

```diff
@@ -0,0 +1,48 @@
+package uk.gov.nationalarchives
+
+import cats.effect.kernel.Async
+import cats.implicits._
+import doobie.Transactor
+import doobie.implicits._
+import doobie.util.transactor.Transactor.Aux
+import uk.gov.nationalarchives.DatabaseConfig.DatabaseCredentials
+import uk.gov.nationalarchives.PuidRepository.{AllPuidInformation, PuidInformation}
+
+class PuidRepository[F[_]: Async](xa: Aux[F, Unit]) {
+
+  private def allowedPuids(): F[List[PuidInformation]] = {
+    val query = sql"""SELECT "PUID", 'NonJudgmentFormat'::text FROM "AllowedPuids"; """.query[PuidInformation].to[List]
+    query.transact(xa)
+  }
+
+  private def disAllowedPuids(): F[List[PuidInformation]] = {
+    val query = sql"""SELECT "PUID", "Reason" FROM "DisallowedPuids"; """.query[PuidInformation].to[List]
+    query.transact(xa)
+  }
+
+  def allPuids: F[AllPuidInformation] = for {
+    allowedPuids <- allowedPuids()
+    disallowedPuids <- disAllowedPuids()
+  } yield AllPuidInformation(allowedPuids, disallowedPuids)
+
+}
+
+object PuidRepository {
+  case class AllPuidInformation(allowedPuids: List[PuidInformation], disallowedPuids: List[PuidInformation])
+
+  case class PuidInformation(puid: String, reason: String)
```

Comment:
There is an additional column in the `DisallowedPuids` table called `Active`. This should be used to check whether the particular puid should cause a rejection of the transfer. If the "active" value is `false` then the disallowed puids should be ignored.

See the code here: https://github.com/nationalarchives/tdr-consignment-api/blob/master/src/main/scala/uk/gov/nationalarchives/tdr/api/service/FFIDMetadataService.scala#L117

Reply:
I've added the active column in, see my other comment.